### PR TITLE
TASK: Display number of validation errors for each tab in inspector

### PR DESCRIPTION
**Pull request by lorissikora**
**What I did**
Completed issue #2377. The number of errors now gets displayed as a badge at the top right corner. This helps to show, why the "Apply" button is disabled when errors are in other tabs and thus not visible.

**How I did it**
The "validationError" object was already available in the redux store. I used it to summarize a count for each tab by testing how many of the field ids, whose values are faulty, are contained inside a tab group. Finally, I created a notification badge for the "Tabs" component, which is displayed if the count is higher than 0.

**How to verify it**
Delete the value of for instance a mandatory field and check if a badge with the number 1 will appear.

Bildschirmfoto 2019-03-19 um 16 53 20

### CODE REVIEW

File: packages/react-ui-components/src/Tabs/tabs.tsx [hunk 1]
```diff
@@ -171,6 +173,11 @@ export interface TabMenuItemProps {
      */
     tooltip?: string;
 
+    /**
+     * An optional notification count displayed as a badge.
+     */
+    notifications?: number;
```
Comment (dimaip) [inline packages/react-ui-components/src/Tabs/tabs.tsx]: I'm curious if adding an extra prop for this is the only way to do it. Maybe we could allow `title` prop to take any elements, and then pass the badge in there. This way the Tabs component would stay more generic. Wdyt?

**Comment by dimaip** [comment]
We started writing an e2e test, will post it here for now:

```
import {adminUser, checkPropTypes} from './../../utils';
import {ReactSelector} from 'testcafe-react-selectors';
import Page from './../../pageModel';

/* global fixture:true */
/* eslint babel/new-cap: 0 */
export const page = new Page();

fixture`Inspector`
    .beforeEach(async t => {
        await t.useRole(adminUser);
        await page.discardAll();
        await page.goToPage('Home');
    })
    .afterEach(() => checkPropTypes());

test('Show number of validation errors for inspector tab', async t => {
    const uriPathSegmentEditor = await page.inspector.getInspectorEditor('uriPathSegment');
    const uriPathSegmentInput = await uriPathSegmentEditor.findReact('TextInput');
    await t.typeText(uriPathSegmentInput, ' ');
    const activeTabMenuItem = await ReactSelector('TabMenuItem').withProps('isActive', true);
    const badge = await activeTabMenuItem.findReact('Badge');
    await t
        .expect(await badge.getReact(({props}) => props.label)).eql('1', 'The badge shows one validation error in Props');
});
```

**Comment by markusguenther** [comment]
@lorissikora Do you want to give the end to end test a try?

File: packages/react-ui-components/src/Tabs/tabs.tsx (hunk 1 above)
Comment (dimaip) [reply packages/react-ui-components/src/Tabs/tabs.tsx]: @markusguenther what's your opinion about this?

Comment (markusguenther) [reply packages/react-ui-components/src/Tabs/tabs.tsx]: Yeah maybe it is more elegant to passthru the Badge to the title. If you need help with that just ping us @lorissikora  :)

**Comment by markusguenther** [comment]
Hi @lorissikora,

could we help you with the PR. It is your first one in the neos project and we really
appreciate that. We can help you out or if you want take over.

Would be nice if you can ping us :)

**Comment by lorissikora** [comment]
Hi, sorry for my lack of communication. Been "out of office" for two weeks and I'm still figuring out the e2e test. Will give it another shot.

**Comment by markusguenther** [comment]
Awesome, if you need help just ask or ping us on slack :)

We have the #neos-ui channel :)

**Comment by markusguenther** [comment]
Hi @lorissikora,

how are you and how it is going. Do you have questions or need help?
Feel free to contact us :)

**Comment by markusguenther** [comment]
Hi @lorissikora are you still on it. Do you need some helping hands?

**Comment by lorenzulrich** [comment]
Please take over this commit from Loris. He's currently not working with Neos at all. He wanted to finish this, but had trouble with the tests so he didn't manage to finish while still working with Neos.

**Comment by markusguenther** [comment]
Would be nice to have for 5.1 as well or? @davidspiola 
Could check that out but that would shift the work on the emberjs part.

**Comment by davidspiola** [comment]
I added it to the releas board, thx @markusguenther

**Comment by markusguenther** [comment]
Could not adjust the PR for some reason.
Replacement is https://github.com/neos/neos-ui/pull/2589

**Pull request closed without merging**